sequence.go

```diff
@@ -92,3 +92,77 @@ func getFeatureSequence(feature Feature, location Location) string {
 
 	return sequenceString
 }
+
+func cartRune(inList ...[]rune) [][]rune {
+	// An iteratitive approach to calculate Cartesian product of two or more lists
+	// Adapted from https://rosettacode.org/wiki/Cartesian_product_of_two_or_more_lists
+	// supposedly "minimizes allocations and computes and fills the result sequentially"
+
+	var possibleVariants int = 1 // a counter used to determine the possible number of variants
+	for _, inList := range inList {
+		possibleVariants *= len(inList)
+	}
+	if possibleVariants == 0 {
+		return nil // in the future this could be part of an error return?
+	}
+	allVariants := make([][]rune, possibleVariants)              // this is the 2D slice where all variants will be stored
+	variantHolders := make([]rune, possibleVariants*len(inList)) // this is an empty slice with a length totaling the size of all input characters
+	variantChoices := make([]int, len(inList))                   // these will be all the possible variants
```

Comment:
Why are these allocations necessary? Can you run them without?

Reply:
These allocations are helping to reduce the total memory footprint. Full disclosure: this is a retool of the cartesian product from Rosetta code https://rosettacode.org/wiki/Cartesian_product_of_two_or_more_lists